macros/src/lib.rs

```diff
@@ -1039,9 +1050,25 @@ pub fn write(ts: TokenStream) -> TokenStream {
     .into()
 }
 
+/// work around restrictions on length and allowed characters imposed by macos linker
+/// returns (note the comma character for macos):
+///   under macos: ".defmt," + 16 character hex digest of symbol's hash
+///   otherwise:   ".defmt." + prefix + symbol
+fn mksection(prefix: &str, symbol: &str) -> String {
+    let mut sub_section = format!(".{}{}", prefix, symbol);
+
+    if cfg!(target_os = "macos") {
```

Comment:
These reflect the system that the macro crate is built for, which is always the host system. We either need to check the `TARGET` env var (if it is even set), or emit `link_section` attributes for all target platforms at once, gated with `cfg_attr`.

Reply:
…whoops. fixed?

Reply:
Yeah, seems good! We should add a test like described in https://github.com/knurling-rs/defmt/issues/336#issuecomment-761094732 though, to make sure this really fixed the bug and to prevent regressions.

Reply:
already messaged Jorge about this 🙃